acme/acme/client.py

```diff
@@ -155,7 +156,23 @@ def answer_challenge(self, challb, response):
         :raises .UnexpectedUpdate:
 
         """
-        response = self._post(challb.uri, response)
+        # Because sending keyAuthorization in a response challenge is deprecated,
+        # we let KeyAuthorizationResponseChallenge not dumping it (default behavior).
+        # However as a migration path, we temporarily expect a malformed error from the server,
+        # and fallback by resending the challenge response with the keyAuthorization field.
+        # TODO: Remove this fallback for Certbot 0.34.0
+        try:
+            response = self._post(challb.uri, response)
+        except messages.Error as error:
+            if (error.code == 'malformed'
+                    and isinstance(response, challenges.KeyAuthorizationChallengeResponse)):
+                logger.debug('Error while responding to a challenge without keyAuthorization '
+                             'in the JWS, your ACME CA server may not support it:\n%s', error)
+                logger.debug('Retrying request with keyAuthorization set.')
+                response.dump_authorization_key(True)
+                response = self._post(challb.uri, response)
+            else:  # pragma: no cover
+                raise
```

Comment:
nit: Adding a test to get coverage on this line seems trivial. Any reason not to add it?

Reply:
I even did the negative test to check that branch! Enabling the coverage here.